### DISCUSSION TITLE
Wyze Bulb - issues setting color temp?

### DISCUSSION THREAD

**Discussion by reidprichard** [SecKatie/ha-wyzeapi#239] [Q&A] [answered]
I'm pretty techy, but I'm new to Home Assistant, so I could just be missing something. I configured Home Assistant OS (v4.20; newer has issues) on an old PC, and I was able to get Home Assistant to recognize my 8 Wyze Bulbs (not the RGB ones). However, I cannot automate color temperature. I can manually control it with the sliders in the app, but my only automation actions available are: "turn off", "turn on", "toggle", "increase brightness", and "decrease brightness". Do changes to color temp need to be implemented a different way (e.g. a .yaml file), or is this a bug?

I'm also finding that changes to color/brightness for my group lights rarely succeed on all 8 - I usually have to fidget with the slider a few more times to get all the lights to change.

**Comment by SecKatie** [comment, marked as answer]
Hmm. I don't think I have ever set color temp automatically but there should be an option in the light turn on service to set the color temp. I believe it is through some optional data. I would recommend looking at home assistant documentation for that piece.

On the issue setting it for a group of lights I replied to your issue so please take a look at that response for my best guess of what is happening

**Reply by reidprichard** [reply]
I'm clearly just getting the hang of GitHub as well - thanks for your patience, and feel free to let me know if I post something in the wrong place, not following convention, etc.

Semi-related to my bug report, I am creating a custom Python service to change the color/brightness of any lights that previously failed to update. I accomplish this by comparing the current brightness/color to the desired for each light, then only updating the bad ones. 

I'm seeing strange behavior with the color temp, which is breaking my script. The color temp reported by the light doesn't quite match the value I set. For example, firing light.turn_on with color_temp = 200 results in the device reporting a value of 214. Similarly, 300 results in 338, and 400 results in 461. In addition to messing up my script, this also makes it difficult to set the lights to the warmest or coldest temps. They report min and max mireds of 153 and 500, but setting color to anything close to either value doesn't do anything (because a commanded value of 500 results in an actual value out of range).

Am I doing something wrong, or is this a known issue with Wyze?

**Reply by hellboyfishman** [reply]
I don't know if others have this issue as well, but I do.  

I can call the on/off service in Node Red for my bulbs, but setting the color of the RGB or the color temperature of my other bulbs never works correctly.   If I split it into 2 service calls it works about 20% on the time but almost never on the RGB.  I've also noticed that it doesn't update the state of the bulb unless I move the slider in the app.  I have one bulb I just turn on and off in HA and for weeks it shows that the color is "green" on my dashboard.   

I was kind of under the assumption that it had to do with the rate limiting and was hoping it would get better but so far, no dice.

**Comment by reidprichard** [comment]
Thanks! I realized I had a fundamental misunderstanding of how scripting works - I thought changing brightness or color would be a separate service from just turning the lights on. I got it working.

**Comment by SecKatie** [comment]
Hmm this might be a bug in the integration. Worth investigation. I will check on it tomorrow

**Comment by JoeSchubert** [comment]
This has been corrected.
